src/truncated/loguniform.jl

```diff
@@ -0,0 +1,7 @@
+truncated(d::LogUniform, lo::T, hi::T) where {T<:Real} = truncated_LogUniform(d,lo,hi)
+truncated(d::LogUniform, lo::Integer, hi::Integer) = truncated_LogUniform(d,lo,hi)
+
+function truncated_LogUniform(d::LogUniform, lo, hi)
+    a,b = params(d)
+    LogUniform(max(a, lo), min(b, hi))
+end
```

Comment:
We only need the following:
```suggestion
function truncated(d::LogUniform, lo::T, hi::T) where {T<:Real}
    a, b = params(d)
    return LogUniform(max(a, lo), min(b, hi))
end
```

Reply:
Mhh I got ambiguities with your suggestion. The problem is 
https://github.com/JuliaStats/Distributions.jl/blob/master/src/truncate.jl#L42

Reply:
Ah well spotted, I missed the definition for integers (I am not sure if it's a good default, maybe it should be removed in a separate PR). In any case, can you check if the following works?
```suggestion
function truncated(d::LogUniform, lo::T, hi::T) where {T<:Real}
    a, b = params(d)
    return LogUniform(max(a, lo), min(b, hi))
end

# fix method ambiguity
truncated(d::LogUniform, lo::Integer, hi::Integer) = truncated(d, float(lo), float(hi))
```
This would basically be the same as the definition for `Uniform`.

Reply:
Mhhh it has the following unexpected (to me) consequence:
```
truncated(LogUniform(2, 10), 2, 14) === LogUniform(2.0, 10.0)
```
We started with integers and ended up with floats for no good reason.

Reply:
Ah, this is annoying but expected... Ideally we would just remove the default method for `Integer`s in src/truncated.jl but it's a bit unrelated to this PR and might break things. I'll check and see if I can fix it in a separate PR.

Reply:
It seems to work, I made a PR: https://github.com/JuliaStats/Distributions.jl/pull/1419

Reply:
The PR was merged and released. If you merge the latest changes from the master branch, `truncated(d::LogUniform, lo::T, hi::T)` should be sufficient.

Reply:
Works!